### ISSUE TITLE
Few improvement suggestions for the plugin 

### ISSUE THREAD

**Issue by davidsaada** [stcr/subscribe-to-comments-reloaded#711]
Hello,
Opened this as a ticket, but posting here a bunch of improvement suggestion from someone who just had your plugin function properly, but needed to spend a couple of days to have it work. Please feel free to close this ticket, split it to subtickets or whatever you see right.

- Subscription management: As in many previous tickets here, I got a 404 error when trying to operate it. Only way I could understand how to have it work was by searching this issue here and reading your replies to the tickets. I think it should be explained more thoroughly in the management page configuration (which isn't the case now). 
- Advanced default option: Right now, one can choose in the configuration between no subscription, subscribe to comments on own comment or to all comments in the post. However, given user feedbacks, this falls short of what they really need: Thing is that some users always want to be subscribed to emails (to own comments or all), while others never want that. Choosing one default cannot satisfy all, which means that users need to set the subscription type in each and every post, which can be quite frustrating. What I suggest here is to add a "Remember last choice" option in the option list (not visible to users, just in plugin configuration), so most users wouldn't practically need to bother about that except for the first time. Alternatively, this can also be a new configuration option. 
- Default admin options: As the admin, the first thing I do when deploying a change would be to test it properly. However, few of the default options for admins stood in my way, such as:
  - Let admins subscribe: Why is this turned off by default? Admins are usually just like any other users, just with extra rights. But more important, if one wants to test how new stuff works, it would be better that the behavior would be as similar to the regular users' one.
  - Send a hidden copy of the comments to management: This practically drove me crazy - I thought that the fact I was receiving mails for all comments was a misconfiguration, until I found this was enabled by default. In addition, there's no real reason why management should be spammed with all these mails by default.
- The ability to manage the subscriptions of this post by an admin is also quite misleading. There's no indication that this ability is only given to admins. So when I first tried that and saw all the mail addresses of the users subscribed to the post, I immediately disabled the plugin and even considered uninstalling it. Only after suspecting this was the reason, I created a regular test user just to test this. I believe that a proper indication in the text there, such as "(admin only)", would resolve this issue.
- Own comments won't trigger the subscription mails. This makes sense of course, but I wasn't aware of that when testing the plugin, so when adding test comments and not getting mails, I thought (yet again) the plugin wasn't working properly. Maybe it's worth adding a config option to enable that for test purposes (which will also show the admin that they won't get emails for that normally). 
- Enable/disable buttons in the configuration are counter-intuitive, in the sense that it's not clear whether the option is enabled or disabled. I guess that traditional radio buttons or drop-down lists would be more intuitive. 
- Hebrew translation has a few flaws. Can post them in a separate ticket or even create a PR to fix them if you wish.

Sorry if this was too much. I really like your plugin and all in all it satisfies my needs. Problem is that I needed to spend a lot of effort (accompanied by a lot of frustration, which initially made me knock few stars in the rating) to have it work. I believe that fixing those issues would be beneficial for others as well.

Cheers,
David.

**Comment by BobaWebDev** [comment]
Hi @davidsaada 

Thank you for the detailed suggestions. I will pass this on to the plugin owner for feedback.
